### PR TITLE
dev-util/idea-ultimate:15 - fix file-collisions plus some small improvements

### PR DESCRIPTION
**Pull request by tomboy-64**
Bugs:
- slot-collisions for :14 and :15 (does idea really need slotting?)
- update to latest version (it's EAP and needs regular updates or ultimate stops working -> remove 15 altogether?)
- quoted ${dir} in line 13
- adjusted inotify limit as per [Jetbrains' advice](https://confluence.jetbrains.com/display/IDEADEV/Inotify+Watches+Limit) in line 36

### CODE REVIEW

**Comment by switch87** [comment]
can you add the fixes to the existing ebuilds (also 14)?
Isn't it better to create /etc/sysctl.d/99-sysctl.conf and first check if the setting is still the original, considering it is not idea-specific and it could be already altered for other purposes.

**Comment by tomboy-64** [comment]
> Isn't it better to create /etc/sysctl.d/99-sysctl.conf and first check if the setting is still the original, considering it is not idea-specific and it could be already altered for other purposes. 
Sure. Feel free to implement it.

Apart from that, your revision bump is ineffective. Even with a valid license, EAP-versions (e.g. 15...pre) are only valid for a month. And as soon as a new version is released and IDEA notices this over the Internet, it stops working and demands being upgraded. E.g. if 15-pre is supposed to stay in the tree, it needs updates ~every 2 weeks.

**Comment by monsieurp** [comment]
folks, which PR should I merge? this one or PR #119?

**Comment by tomboy-64** [comment]
Alright, revamped the ebuilds.
- all go back to SLOT="0"
- _pre versions do not contain any keywords (since EAP code quality is utterly catastrophic)
- removed older 15.0.0 ebuilds, since those actually don't work anymore.

What do you think, Gert?

**Comment by switch87** [comment]
I can't test them until tomorrow, but looks good. I am closing my pull request.

File: dev-util/idea-ultimate/idea-ultimate-14.1.5.141.2735.5_pre.ebuild [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+MY_PV="$(get_version_component_range 1-3)" # Always name EAP-versions with '_pre' for clarity!
+MY_PN="idea"
+
+DESCRIPTION="A complete toolset for web, mobile and enterprise development"
+HOMEPAGE="http://www.jetbrains.com/idea"
+SRC_URI="https://download.jetbrains.com/idea/${MY_PN}IU-${MY_PV}.tar.gz"
+
+LICENSE="IDEA IDEA_Academic IDEA_Classroom IDEA_OpenSource IDEA_Personal"
+IUSE=""
+KEYWORDS="" # No keywords for EAP versions. Code quality sucks.
```
Comment (switch87) [inline dev-util/idea-ultimate/idea-ultimate-14.1.5.141.2735.5_pre.ebuild]: why no keywords? ...

File: dev-util/idea-ultimate/idea-ultimate-15.0.0.142.5047.6_pre.ebuild [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+MY_PV="$(get_version_component_range 1-3)" # Always name EAP-versions with '_pre' for clarity!
+MY_PN="idea"
+
+DESCRIPTION="A complete toolset for web, mobile and enterprise development"
+HOMEPAGE="http://www.jetbrains.com/idea"
+SRC_URI="https://download.jetbrains.com/idea/${MY_PN}IU-${PV_STRING}.tar.gz"
+
+LICENSE="IDEA IDEA_Academic IDEA_Classroom IDEA_OpenSource IDEA_Personal"
+IUSE=""
+KEYWORDS="" # No keywords for EAP versions. Code quality sucks.
```
Comment (switch87) [inline dev-util/idea-ultimate/idea-ultimate-15.0.0.142.5047.6_pre.ebuild]: idem

File: dev-util/idea-ultimate/idea-ultimate-14.1.5.141.2735.5_pre.ebuild (hunk 1 above)
Comment (tomboy-64) [inline dev-util/idea-ultimate/idea-ultimate-14.1.5.141.2735.5_pre.ebuild]: Have you actually tried the EAP releases? They are ridiculously buggy. More like alpha-versions. Someone who wants to work with it should actually be *VERY* sure.

Comment (switch87) [inline dev-util/idea-ultimate/idea-ultimate-14.1.5.141.2735.5_pre.ebuild]: devs who require their IDE to function reliable do not use pre-release versions (although the actual programming-code is not influenced by idea), they should have the testing keywords. I did a request for stabilization for the stable version. [561294](https://bugs.gentoo.org/show_bug.cgi?id=561294)

Comment (tomboy-64) [inline dev-util/idea-ultimate/idea-ultimate-14.1.5.141.2735.5_pre.ebuild]: Okay.
But I'm gonna put a conditional info at the end of the ebuilds.

MonsieurP will merge it then; he asks for you to close your request.

**Comment by tomboy-64** [comment]
Implemented the changes.
While adding stuff to QA_WX_LOAD for idea-ultimate-15.0.0... however, i noticed it somehow still complains about fs-notifier-arm.
Also, for the same ebuild, it complains that a lot of .so can't be stripped (because they are for different arches, arm and solaris). Suggestions how to handle this?

**Comment by tomboy-64** [comment]
Alright, considering it usable.
Agreed, Gert?

**Comment by mgorny** [comment]
@gentoo/proxy-maint

**Comment by tomboy-64** [comment]
Since there are no vetoes I wil request merge by Monday night, european time.

**Comment by switch87** [comment]
If you speak to me, you have to say @switch87 for me to notice it, not Gert. I cant test it until tuesday, you added so much I cant see if its ok or not.
You have to reset and recommit, commit messenges are not like they should be and will be declined.

**Comment by tomboy-64** [comment]
@switch87 sure thing. I re-based and re-committed.

**Comment by switch87** [comment]
Tested it, for me it´s ok :+1: 
@monsieurp @gentoo/proxy-maint

**Comment by monsieurp** [comment]
Merged! :+1:

**Pull request merged**